### PR TITLE
Rebuild python39, skipping some optional test dependencies

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

- includes #113
- adds a `building_on_new_platform`variable, which hides historically-problematic-during-migration, yet optional, test-time dependencies

### CODE REVIEW

**Comment by bollwyvl** [comment]
Gah, messed up my selector... trying again locally...

**Comment by bollwyvl** [comment]
Lots of :green_circle:, but there is no victory until windows works...

**Comment by jakirkham** [comment]
Looks like everything passed! 🎉 🟢 

Not sure why Travis CI shows up twice (once as passing and once as pending). Saw the same thing on another build earlier. Maybe some hiccup between Travis CI and GitHub's API. Anyways I think we can safely ignore it 😉

File: recipe/meta.yaml [hunk 1]
```diff
@@ -37,25 +39,27 @@ requirements:
     - setuptools >=18.5
 
 # generally failing
-{% set nose_exclude = "recursion" %}
+{% set nose_exclude = "recursion|test_set_matplotlib_formats|test_set_matplotlib_close" %}
```
Comment (bollwyvl) [inline recipe/meta.yaml:42]: actually, we could probably put these back, as it will nicely skip them if matplotlib is missing... can be another PR...

File: recipe/meta.yaml [hunk 2]
```diff
@@ -37,25 +39,27 @@ requirements:
...
     - nbformat
     - nose >=0.10.1
-    - numpy
-    - pygments
+    - pip
     - requests
     - testpath
+    # would _ideally_ be in the block below, but messes with re-rendering
+    - numpy
+    {% if not building_on_new_platform %}
```
Comment (ocefpaf) [inline recipe/meta.yaml:57]: We need to check if we can do test downstreams instead: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#downstream-tests

**Comment by bollwyvl** [comment]
Sure, would be appropriate for ipykernel, but probably not for mpl/ioloopy
things...
